### ISSUE TITLE
HNC: v0.3.0 doesn't work

### ISSUE THREAD

**Issue by zimisu** [kubernetes-sigs/multi-tenancy#777]
I tried to install HNC v0.3.0, but the hnc-controller-manager pod always crash without any message
```
$ kubectl get po -n hnc-system
NAME                                      READY   STATUS             RESTARTS   AGE
hnc-controller-manager-5956dcd796-lmnpm   1/2     CrashLoopBackOff   239        20h
$ kubectl logs -n hnc-system hnc-controller-manager-5956dcd796-lmnpm manager
----- nothing printed -----
$ kubectl logs -n hnc-system hnc-controller-manager-5956dcd796-lmnpm kube-rbac-proxy
I0527 10:07:55.532579       1 main.go:209] Generating self signed cert as no cert is provided
I0527 10:07:55.652383       1 main.go:242] Listening securely on 0.0.0.0:8443
$
```
Is the image `gcr.io/k8s-staging-multitenancy/hnc/controller:v0.3.0` correct?

**Comment by adrianludwin** [comment]
Yes, that's the correct image. Did you have Jetstack cert-manager installed?

**Comment by adrianludwin** [comment]
Hi @zimisu , do you want to try [HNC v0.4.0 rc1](https://github.com/kubernetes-sigs/multi-tenancy/releases/tag/hnc-v0.4.0-rc1)? It has a simplified installation so may work better for you. The "final" version of 0.4.0 will be out early next week.

**Comment by zoetrope** [comment]
I faced the same problem with HNC v0.3.0 and v0.4.0-rc1.
There is no message, so I don't know the cause.

I'm using Kind v0.7.0 and Kubernetes v1.17.0.

**Comment by adrianludwin** [comment]
I know there's a problem on Kind which I've been meaning to look into
(#770). Do you see the same problem on a "real" Kubernetes installation?

**Comment by zimisu** [comment]
The cert-manager was installed, v0.2.0 works well in same environment. I'll try v0.4.0-rc1, thanks

**Comment by zimisu** [comment]
> I know there's a problem on Kind which I've been meaning to look into (#770). Do you see the same problem on a "real" Kubernetes installation?

yes,  the problem still exist on my single node k8s cluster(v1.17.6, install with kubeadm, not kind) with hnc-v0.3.0 and hnc-v0.4.0-rc1

**Comment by adrianludwin** [comment]
Ok, thanks for the report. Are there no other logs from the container?

Are you able to manually pull the container images and start them locally
(using `docker pull` and `docker run`)?

**Comment by adrianludwin** [comment]
Per #770, the most likely culprit here is the Stackdriver exporter, which (I'm guessing) errors out if it's not on GKE. I'll fix that in 0.4.0-rc2.

**Comment by adrianludwin** [comment]
Yup, it was the Stackdriver exporter. I'll have 0.4.0-rc2 published shortly. My apologies.

Do you need me to fix 0.3.0 as well, or are you willing to use 0.4?

**Comment by adrianludwin** [comment]
This is a duplicate of #770, closing this copy. Please add new comments there.

/close
